railties/test/application/rake_test.rb

```diff
@@ -40,7 +40,7 @@ def test_not_protected_when_previous_migration_was_not_production
       with_rails_env "test" do
         rails "generate", "model", "product", "name:string"
         rails "db:create", "db:migrate"
-        output = Dir.chdir(app_path) { rails("db:test:prepare", "test") }
+        output = rails("db:test:prepare", allow_failure: true)
```

Comment:
Is this currently raising? What failure do we need to allow? :confused:

Reply:
If for some reason `db:test:prepare` fails, the code below
```
refute_match(/ActiveRecord::ProtectedEnvironmentError/, output)
```
 won't be executed

instead we will see error
````
RuntimeError: rails command failed (1): bin/rails db:test:prepare 2>&1
````

Reply:
Why would that be a bad thing?

If the command fails, I want the test to fail, not pass just because the failure doesn't contain that string.

Reply:
Because if leave it as it is and `db:test:prepare` fails for some reason(for example: adding bad changes in a PR)

this line will never be executed
```
refute_match(/ActiveRecord::ProtectedEnvironmentError/, output)
```
so if `db:test:prepare` raise `ActiveRecord::ProtectedEnvironmentError` or any antother error, test status will be as `error` but not as `failure` or `passed`.

In this way, this test doesn't check what we expect.

Reply:
If there is an error in that command, then an error has occurred, and so 'error' would be the correct result for the test.

Reply:
@matthewd I just removed the changes